tests/trezor/trezor_tests.cpp

```diff
@@ -797,85 +1000,106 @@ bool gen_trezor_base::generate(std::vector<test_event_entry>& events)
+    {addr_alice_sub_0_2, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_0_4, true, MK_TCOINS(0.1)}}), 1, CUR_HF);
 
   // Transfer to other accounts
-  MAKE_TX_MIX_DEST_LIST_RCT(events, txs_blk_5, m_miner_account, build_dsts(addr_alice_sub_1_0, true, MK_COINS(1) >> 1), 10, blk_4);
-  MAKE_TX_MIX_DEST_LIST_RCT(events, txs_blk_5, m_miner_account, build_dsts(addr_alice_sub_1_1, true, MK_COINS(1) >> 1), 10, blk_4);
-  MAKE_TX_MIX_DEST_LIST_RCT(events, txs_blk_5, m_miner_account, build_dsts({{addr_alice_sub_1_0, true, MK_COINS(1) >> 1}, {addr_alice_sub_1_1, true, MK_COINS(1) >> 1}, {addr_alice_sub_0_3, true, MK_COINS(1) >> 1}}), 10, blk_4);
-  MAKE_TX_MIX_DEST_LIST_RCT(events, txs_blk_5, m_miner_account, build_dsts({{addr_alice_sub_1_1, true, MK_COINS(1) >> 1}, {addr_alice_sub_1_1, true, MK_COINS(1) >> 1}, {addr_alice_sub_0_2, true, MK_COINS(1) >> 1}}), 10, blk_4);
-  MAKE_TX_MIX_DEST_LIST_RCT(events, txs_blk_5, m_miner_account, build_dsts({{addr_alice_sub_1_2, true, MK_COINS(1) >> 1}, {addr_alice_sub_1_1, true, MK_COINS(1) >> 1}, {addr_alice_sub_0_5, true, MK_COINS(1) >> 1}}), 10, blk_4);
-
-  // Simple RCT transactions
-  MAKE_TX_MIX_LIST_RCT(events, txs_blk_5, m_miner_account, m_alice_account, MK_COINS(7), 10, blk_4);
-  MAKE_TX_MIX_LIST_RCT(events, txs_blk_5, m_miner_account, m_alice_account, MK_COINS(10), 10, blk_4);
-  MAKE_TX_MIX_LIST_RCT(events, txs_blk_5, m_miner_account, m_alice_account, MK_COINS(30), 10, blk_4);
-  MAKE_TX_MIX_LIST_RCT(events, txs_blk_5, m_miner_account, m_alice_account, MK_COINS(40), 10, blk_4);
-  MAKE_NEXT_BLOCK_TX_LIST_HF(events, blk_5, blk_4r, m_miner_account, txs_blk_5, CUR_HF);
+  generate_tx_block(events, txs_blk_5, last_block, m_miner_account, build_dsts({
+    {addr_alice_sub_1_0, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_1_1, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_1_0, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_1_1, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_0_3, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_1_1, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_1_1, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_0_2, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_1_2, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_1_1, true, MK_TCOINS(0.1)},
+    {addr_alice_sub_0_5, true, MK_TCOINS(0.1)}}), 1, CUR_HF);
+  MAKE_NEXT_BLOCK_TX_LIST_HF(events, blk_5, last_block, m_miner_account, txs_blk_5, CUR_HF);
+  last_block = blk_5;
 
   // Simple transaction check
   bool resx = rct::verRctSemanticsSimple(txs_blk_5.begin()->rct_signatures);
   bool resy = rct::verRctNonSemanticsSimple(txs_blk_5.begin()->rct_signatures);
   CHECK_AND_ASSERT_THROW_MES(resx, "Tsx5[0] semantics failed");
   CHECK_AND_ASSERT_THROW_MES(resy, "Tsx5[0] non-semantics failed");
 
-  REWIND_BLOCKS_HF(events, blk_5r, blk_5, m_miner_account, CUR_HF);  // rewind to unlock
+  // Rewind blocks so there are more spendable coins
+  last_block = rewind_blocks_with_decoys(events, last_block, CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE + 2, CUR_HF, TEST_DECOY_MINING_COUNTS);
```

Comment:
`+ 2` was added in the recent commits, is it to pop more transactions into mempool? The other `rewind_blocks_with_decoys` further down doesn't have this `+ 2`.

Reply:
correct, I had to increase number of transactions to pick from as the last test was failing, so there is enough buffer for that

Reply:
@SChernykh can I pls resolve this one?

Reply:
Yes, and other unresolved comments too.